### DISCUSSION TITLE
How to expose a substream from Frigate to HA

### DISCUSSION THREAD

**Discussion by daernsinstantfortress** [blakeblackshear/frigate#9317] [Ask A Question]
I'd like to be able to access both the main and substream cameras from HA for a single Frigate camera e.g.

camera.frontdoor_main
camera.frontdoor_sub

Both are exposed to Frigate via go2rtc, but is there any way to see both camera entities in HA?

(The reason I want this is that the substream has significantly less latency so is better for doing snapshots than the main stream, which is too laggy to give accurate results, but I still want to use the main stream for most other uses cases)

**Comment by NickM-27** [comment]
Frigate card supports sub streams

Otherwise you'd need to add the second one as a generic camera entity

**Reply by daernsinstantfortress** [reply]
Thanks Nick.

Oddly, I can't add my Frigate-configured go2rtc stream as a generic camera. I'm using `rtsp://127.0.0.1:8554/front_doorbell_camera_sub`  with TCP and no auth, but the test image always comes back broken, and I cannot see video through the created entity if I accept it:

![image](https://github.com/blakeblackshear/frigate/assets/7900720/6bca10d9-9df6-4570-a222-bbc90d5edea6)

I've added the RTSP restream port to the Frigate add-on config, so a bit confused here. This is the same URL as I use in my Frigate config. I've also tried this URL with my HA IP address, but which also doesn't work as a generic camera, despite opening fine in VLC.

Any clues?

**Reply by NickM-27** [reply]
you can't use 127.0.0.1, that is localhost and go2rtc is not exposed on home assistants localhost

**Reply by daernsinstantfortress** [reply]
Thanks - I'll keep playing around here.
